lib/datasets_test.go

```diff
@@ -117,14 +117,7 @@ func TestDatasetRequestsSave(t *testing.T) {
 		err         string
 	}{
 
-		{"empty params", SaveParams{}, "name or bodypath is required"},
-		// {&dataset.Dataset{Peername: "foo", Name: "bar"}, nil, "error with previous reference: error fetching peer from store: profile: not found"},
-		// {&dataset.Dataset{Peername: "bad", Name: "path", Commit: &dataset.Commit{Qri: "qri:st"}}, nil, "decoding dataset: invalid commit 'qri' value: qri:st"},
-		// {&dataset.Dataset{Peername: "bad", Name: "path", BodyPath: "/bad/path"}, nil, "error with previous reference: error fetching peer from store: profile: not found"},
-		// {&dataset.Dataset{BodyPath: "testdata/q_bang.svg"}, nil, "invalid data format: unsupported file type: '.svg'"},
-		// {&dataset.Dataset{Peername: "me", Name: "cities", BodyPath: "http://localhost:999999/bad/url"}, nil, "fetching body url: Get http://localhost:999999/bad/url: dial tcp: address 999999: invalid port"},
-		// {&dataset.Dataset{Name: "bad name", BodyPath: jobsBodyPath}, nil, "invalid name: error: illegal name 'bad name', names must start with a letter and consist of only a-z,0-9, and _. max length 144 characters"},
-		// {&dataset.Dataset{BodyPath: jobsBodyPath, Commit: &dataset.Commit{Qri: "qri:st"}}, nil, "decoding dataset: invalid commit 'qri' value: qri:st"},
+		{"empty params", SaveParams{}, "no changes to save"},
```

Comment:
Definitely think this is a better error. However, it reminds me of this problem:
https://github.com/qri-io/qri/blob/master/cmd/save_test.go#L297

We used to allow saving with just a structure and no body. We don't need to necessarily fix that in this PR, but is there a plan for whether the newly refactored save path will allow for this behavior to be restored in the future?

Reply:
yes. This PR shouldn't inhibit no-body saves. Most of that testing needs to occur in `dsfs.CreateDataset`, then get propagated up.